langchain/embeddings/openai.py

```diff
@@ -196,6 +196,8 @@ def _invocation_params(self) -> Dict:
             "api_type": self.openai_api_type,
             "api_version": self.openai_api_version,
         }
+        if self.openai_api_type in ("azure", "azure_ad", "azuread"):
+            openai_args["engine"] = self.deployment
```

Comment:
hey @jacobajit, apologies for the slow review!

just to make sure i understand correctly, "model" doesn't need to be removed here as well?

Reply:
Correct. For Azure purposes, `model` is optional but harmless when used in addition to `engine` (or `deployment-id` which is the same thing), as seen here: https://github.com/openai/openai-python#microsoft-azure-endpoints

From testing the behavior, it seems like the model name is just ignored since each deployment has only a single model, but we can leave it in just in case it becomes useful at some point.